ui/src/app/workflows/components/workflows-list/workflows-list.tsx

```diff
@@ -53,15 +67,19 @@ export class WorkflowsList extends BasePage<RouteComponentProps<any>, State> {
             managedNamespace: false,
             namespace: this.props.match.params.namespace || Utils.getCurrentNamespace() || '',
             selectedPhases: this.queryParams('phase'),
-            selectedLabels: this.queryParams('label')
+            selectedLabels: this.queryParams('label'),
+            selectedWorkflows: {},
+            batchActionDisabled: {...allBatchActionsEnabled}
         };
     }
 
     public componentDidMount(): void {
         this.fetchWorkflows(this.state.namespace, this.state.selectedPhases, this.state.selectedLabels, this.state.pagination);
+        this.setState({selectedWorkflows: {}});
```

Comment:
Why do we need to set this here? Why is it not enough to just do so in the constructor?

Reply:
I added this because when changing filters, labels or pagination, there was a memory leak due to workflows that were out of view still being held in the `selectedWorkflows` state. I can't remember what the exact console error was but I can find out